### ISSUE TITLE
欧拉角检测不准， angle为检测，angle_gt为真实数据

### ISSUE THREAD

**Issue by a361251388** [polarisZhao/PFLD-pytorch#38]
angle = tensor([[-0.0873,  0.7688, -0.2588]], device='cuda:0')
angle_gt = tensor([[ 88.0343, -60.7853, -62.0728]], device='cuda:0')
angle = tensor([[-0.1598,  0.5773, -0.4115]], device='cuda:0')
angle_gt = tensor([[ 66.4287, -53.0196, -49.0985]], device='cuda:0')
angle = tensor([[-0.0510,  0.7076, -0.3719]], device='cuda:0')
angle_gt = tensor([[ 48.0916, -45.6873, -17.9782]], device='cuda:0')
angle = tensor([[ 0.0888,  0.7797, -0.3578]], device='cuda:0')
angle_gt = tensor([[ 54.0094, -37.7794, -10.2092]], device='cuda:0')
angle = tensor([[-0.0731,  0.8048, -0.0279]], device='cuda:0')
angle_gt = tensor([[15.3467,  3.7500, 10.6802]], device='cuda:0')
angle = tensor([[ 0.0906,  0.6620, -0.2479]], device='cuda:0')
angle_gt = tensor([[ 79.5670, -63.5313, -39.0085]], device='cuda:0')

**Comment by polarisZhao** [comment]
这是弧度制和角度值的问题吧。 不过 solvePnP 这种方法来估计角度确实不大准，但是论文也是这么做。

**Comment by futureisatyourhand** [comment]
> 这是弧度制和角度值的问题吧。 不过 solvePnP 这种方法来估计角度确实不大准，但是论文也是这么做。

hello，I want to know how to issue the problem?  why does this happen?
Thanks!

**Comment by wyfdodogod** [comment]
> > 这是弧度制和角度值的问题吧。 不过 solvePnP 这种方法来估计角度确实不大准，但是论文也是这么做。
> 
> hello，I want to know how to issue the problem? why does this happen? Thanks!

hi beauty,
check his loss function u might know why, "weight_angle = torch.sum(1 - torch.cos(angle - euler_angle_gt), axis=1) "# funny  np.cos(100-(100+n*2*np.pi))=1

**Comment by wyfdodogod** [comment]
> > 这是弧度制和角度值的问题吧。 不过 solvePnP 这种方法来估计角度确实不大准，但是论文也是这么做。
> 
> hello，I want to know how to issue the problem? why does this happen? Thanks!

let me be more specific, (79.5670-0.0906)/3.1415926=25.0086
